### PR TITLE
executor: fix data race in TestSlowQueryOOM

### LINKED ISSUE #33982: data race in infoschema TestSlowQueryOOM

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #33982

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Still data race @crazycs520 
Try `make ut X='run infoschema TestSlowQueryOOM --race'` to verify this change.

File: infoschema/cluster_tables_test.go [hunk 1]
```diff
@@ -701,7 +701,7 @@ select * from t1;
 		checkFn(quota)
 	}
 	for i := 0; i < 100; i++ {
-		quota := rand.Int()%10240 + 1
+		quota := rand.Int()%8192 + 1
```
Comment (tiancaiamao) [inline infoschema/cluster_tables_test.go:704]: unrelated change?

Comment (crazycs520) [reply infoschema/cluster_tables_test.go:704]: I found `10240` maybe too large and won't cause oom in my local test sometime, so I change it to a smaller value.

**Review by tiancaiamao — APPROVED** [review]

File: executor/slow_query.go [hunk 2]
```diff
@@ -198,6 +201,8 @@ func (e *slowQueryRetriever) getPreviousFile() *os.File {
 }
 
 func (e *slowQueryRetriever) parseDataForSlowLog(ctx context.Context, sctx sessionctx.Context) {
+	e.wg.Add(1)
```
Comment (hawkingrei) [inline executor/slow_query.go:204]: ```Add``` should put at the usage of ```parseDataForSlowLog```.


···
e.wg.Add(1)
go e.parseDataForSlowLog(ctx, sctx)
···

BTW can you use the ```util.WaitGroupWrapper``` ？

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**